cmd/apprepository-controller/controller.go

```diff
@@ -403,18 +435,11 @@ func newCronJob(apprepo *apprepov1alpha1.AppRepository) *batchv1beta1.CronJob {
 
 // newSyncJob triggers a job for the AppRepository resource. It also sets the
 // appropriate OwnerReferences on the resource
-func newSyncJob(apprepo *apprepov1alpha1.AppRepository) *batchv1.Job {
+func newSyncJob(apprepo *apprepov1alpha1.AppRepository, kubeappsNamespace string) *batchv1.Job {
 	return &batchv1.Job{
 		ObjectMeta: metav1.ObjectMeta{
-			GenerateName: cronJobName(apprepo) + "-",
-			Namespace:    apprepo.Namespace,
```

Comment:
Yes, because we choose which namespace to write these resources in the actual client call, so if we also write it here, we have to duplicate the logic/args to ensure it is the same (otherwise, the k8s api server rejects it as it should).